test/e2e/react-compiler/reference-library/compiled/README.md

```diff
@@ -0,0 +1,8 @@
+These are manually compiled versions from `src/index`.js with [React Compiler Playground](https://playground.react.dev/) and [Babel.js Repl](https://babeljs.io/repl) (only to compile JSX runtime).
+
+| module                                   | import condition | resolved                             | React Compiler | Server-Client boundary | Valid environment |
+|------------------------------------------|------------------|--------------------------------------|----------------|------------------------|-------------------|
+| `reference-library/client`               | any              | `./compiled/client.js`               | Yes            | Yes                    | client-only       |
+| `reference-library`                      | `react-server`   | `./compiled/index.react-server.js`   | No             | No                     | any               |
+| `reference-library`                      | default          | `./compiled/index.js`   | Yes            | No                     | client-only       |
+| `reference-library/missing-react-server` | any              | `./compiled/missing-react-server.js` | Yes            | No                     | client-only       |
```

Comment:
This is the part most libraries make a mistake because we don't have docs/good error message (error message will be improved with https://github.com/facebook/react/pull/32056). You'd just think you can throw the React Compiler at your library and call it a day. But once this library is supposed to run in a React Server, React will throw because there's nothing to memoize on the React Server.

It's important to understand that the React Compiler currently only targets the browser. It doesn't necessarily produce output optimized for SSR or RSCs.